pilot/pkg/bootstrap/monitoring.go

```diff
@@ -38,7 +38,12 @@ const (
 )
 
 func addMonitor(mux *http.ServeMux) error {
-	exporter, err := ocprom.NewExporter(ocprom.Options{Registry: prometheus.DefaultRegisterer.(*prometheus.Registry)})
+
+	registry := prometheus.NewRegistry()
```

Comment:
This is potentially problematic. Elsewhere in the pilot codebase, I believe there are calls to `prometheus.MustRegister`. IIRC, those calls will use the `DefaultRegisterer` and NOT this `registry`. This could be why you are seeing issues with the dashboard tests.

You might try using the defaultRegistry for registering the collectors (unless you also want to take on converting all of the `prometheus.MustRegister` statements to opencensus `view.Register` style).

Reply:
@douglas-reid Thanks for your instruction. Just update for review.